dds/DCPS/Util.h

```diff
@@ -140,6 +140,35 @@ void push_back(Seq& seq, const typename Seq::value_type& val)
   seq[len] = val;
 }
 
+// Constructs a sorted intersect of the given two sorted ranges [a,aEnd) and [b,bEnd).
+// (for pre-c++17 code for the similar effect of std::set_intersection in c++17)
+template <typename InputIteratorA, typename InputIteratorB, typename OutputIterator>
+OutputIterator intersect_sorted_ranges(InputIteratorA a, InputIteratorA aEnd,
+                                       InputIteratorB b, InputIteratorB bEnd,
+                                       OutputIterator intersect)
+{
+  while (a != aEnd && b != bEnd) {
```

Comment:
This function could delegate to the other one using a LessThan which class < (like std::less but for two different types).
We could also require `A` and `B` to be the same type, but it's probably better to have the same interface as std.